kani-driver/src/call_cbmc.rs

```diff
@@ -164,6 +184,94 @@ impl KaniSession {
     }
 }
 
+impl VerificationResult {
+    /// Computes a `VerificationResult` (kani-driver's notion of the result of a CBMC call) from a
+    /// `VerificationOutput` (cbmc_output_parser's idea of CBMC results).
+    ///
+    /// NOTE: We actually ignore the CBMC exit status, in favor of two checks:
+    ///   1. Examining the actual results of CBMC properties.
+    ///       (CBMC will regularly report "failure" but that's just our cover checks.)
+    ///   2. Positively checking for the presence of results.
+    ///       (Do not mistake lack of results for success: report it as failure.)
+    fn from(output: VerificationOutput, start_time: Instant) -> VerificationResult {
+        if let Some(mut items) = output.processed_items {
+            let result_idx = items.iter().position(|x| matches!(x, ParserItem::Result { .. }));
+            if let Some(result_idx) = result_idx {
+                let result = items.remove(result_idx);
+                if let ParserItem::Result { result } = result {
+                    let number_failed_properties =
+                        result.iter().filter(|prop| prop.status == CheckStatus::Failure).count();
+                    let status = if number_failed_properties == 0 {
+                        VerificationStatus::Success
+                    } else {
+                        VerificationStatus::Failure
+                    };
+                    VerificationResult {
+                        status,
+                        messages: Some(items),
+                        results: Some(result),
+                        exit_status: output.process_status,
+                        runtime: start_time.elapsed(),
+                    }
+                } else {
+                    unreachable!() // We filtered for this to be true
+                }
+            } else {
+                // We never got results from CBMC - also shouldn't really happen unless something went wrong
+                VerificationResult {
+                    status: VerificationStatus::Failure,
+                    messages: Some(items),
+                    results: None,
+                    exit_status: output.process_status,
+                    runtime: start_time.elapsed(),
+                }
+            }
+        } else {
+            // We don't have ANY messages to report - shouldn't really happen unless something went wrong
+            VerificationResult {
+                status: VerificationStatus::Failure,
+                messages: None,
+                results: None,
+                exit_status: output.process_status,
+                runtime: start_time.elapsed(),
+            }
+        }
+    }
+    pub fn mock_success() -> VerificationResult {
+        VerificationResult {
+            status: VerificationStatus::Success,
+            messages: None,
+            results: None,
+            exit_status: 42,
```

Comment:
😂

Make it a constant (e.g., `MOCK_EXIT_STATUS`) and use it here?

Reply:
I didn't think a constant made sense, but I did add comments and audit how they end up used.

This did make me notice that the new results printing should have more exceptions: in addition to `quiet` it also suppresses on `dry-run` and `output-format=old`.